include/boost/detail/winapi/config.hpp

```diff
@@ -54,6 +71,27 @@
 #endif
 #endif
 
+// These constants reflect WINAPI_FAMILY_* macros from winapifamily.h
+#define BOOST_WINAPI_FAMILY_PC_APP      2       // Windows Store: Most Restrictive API Set
+#define BOOST_WINAPI_FAMILY_PHONE_APP   3       // Windows Phone
+#define BOOST_WINAPI_FAMILY_SYSTEM      4       // Windows System and Drivers
+#define BOOST_WINAPI_FAMILY_SERVER      5       // Windows Server
+#define BOOST_WINAPI_FAMILY_DESKTOP_APP 100     // Windows Desktop: Least Restrictive API Set (this value subject to change)
+
+#if !defined(BOOST_USE_WINAPI_FAMILY)
+#if defined(WINAPI_FAMILY)
+#define BOOST_USE_WINAPI_FAMILY WINAPI_FAMILY
+#elif defined(BOOST_WINAPI_IS_MINGW_W64) || (BOOST_WINAPI_WINDOWS_SDK == BOOST_WINAPI_WINDOWS_SDK_8)
+// This is different for Visual Studio 2012 / Windows SDK 8.0 than for later ones
+// MinGW-w64 has the same logic as Windows SDK 8.0
+// See WINAPI_FAMILY_DESKTOP_APP from https://github.com/Alexpux/mingw-w64/blame/master/mingw-w64-headers/include/winapifamily.h
+#define BOOST_USE_WINAPI_FAMILY 0x3 /* == WINAPI_FAMILY_DESKTOP_APP */
+#elif BOOST_WINAPI_WINDOWS_SDK >= BOOST_WINAPI_WINDOWS_SDK_8_1
+// By default if the WINAPI_FAMILY is not defined, select the desktop family
+#define BOOST_USE_WINAPI_FAMILY BOOST_WINAPI_FAMILY_DESKTOP_APP
+#endif
+#endif
```

Comment:
Maybe it's better to include winapifamily.h instead of defining constants depending on the particular SDK version.

Reply:
We can only include winapifamily.h if BOOST_USE_WINDOWS_H is defined, correct?  Further, it can only be included in MinGW-w64 and if the Platform SDK is 8.0 or later.  I actually have this coded up already locally for changes related to boost random so I will pull it back into this.

Reply:
If we really need the family macros then I think it's ok to include winapifamily.h even when `BOOST_USE_WINDOWS_H` is not defined. That header is fairly lightweight and doesn't define any offending macros, so it should be pretty safe.

Reply:
Okay I was trying to avoid that, but I'll simplify.  I struggled with the same decision but I decided initially to follow the pattern in the project.